### PR TITLE
CHE-6545: Add condition for breakpoints

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Ability to add condition for breakpoint.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/6545

Added conditions for breakpoints.

#### Release Notes

Added conditions for breakpoints.

#### Docs PR

not required

### CODE REVIEW

**Comment by tolusha** [comment]
screenshot from 2017-10-17 14-33-02

File: plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/breakpoint/BreakpointContextMenuFactory.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.plugin.debugger.ide.debug.breakpoint;
+
+import org.eclipse.che.api.debug.shared.model.Breakpoint;
+
+/**
+ * Editor tab context menu factory.
+ *
+ * @author Vlad Zhukovskiy
```
Comment (vzhukovs) [inline plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/breakpoint/BreakpointContextMenuFactory.java]: I'm not an author of this interface.

**Review by vzhukovs — APPROVED** [review]

**Comment by tolusha** [comment]
```
[TEST] 	org.eclipse.che.selenium.debugger.CppProjectDebuggingTest.shouldDebugCppProject
[TEST] 	org.eclipse.che.selenium.debugger.CppProjectDebuggingTest.stopDebug
[TEST] 	org.eclipse.che.selenium.debugger.DebugExternalClassTest.shouldDebugJreClass
[TEST] 	org.eclipse.che.selenium.debugger.InnerClassAndLambdaDebuggingTest.shouldDebugMethodLocalInnerClass
[TEST] 	org.eclipse.che.selenium.debugger.InnerClassAndLambdaDebuggingTest.startDebug
[TEST] 	org.eclipse.che.selenium.debugger.InnerClassAndLambdaDebuggingTest.stopDebug
```

**Review by vparfonov — APPROVED** [review]

File: ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/debug/BreakpointRenderer.java [hunk 2]
```diff
@@ -10,44 +10,31 @@
...
-  /**
-   * Add a breakpoint mark in the gutter on the given line.
-   *
-   * @param lineNumber the line where the breakpoint is set
+   * @param breakpoint the given breakpoint
+   * @param active indicates if breakpoint is active or isn't
    * @param action to line numbering changes
    */
-  void addBreakpointMark(int lineNumber, LineChangeAction action);
+  void setMark(Breakpoint breakpoint, boolean active, LineChangeAction action);
```
[the remark this answers is not in the document]
Comment (tolusha) [reply ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/debug/BreakpointRenderer.java]: Agree, my fail.

**Pull request merged**